test/vars/UtilSpec.groovy

```diff
@@ -423,4 +423,17 @@ class UtilSpec extends JenkinsPipelineSpecification {
         then:
         true
     }
+    def "[util.groovy] getNextSnapshot"() {
+        when:
+        def snapshotVersion = groovyScript.getNextSnapshot('0.12.0')
+        then:
+        snapshotVersion == '0.12.1-SNAPSHOT'
+        
+    }
+    def "[util.groovy] getNextSnapshot"() {
+        when:
+        def checkForFormatError = groovyScript.getNextSnapshot('a.12.0')
```

Comment:
The length of the split array is still 3 here, so this doesn't check the error you want. But, the test also throws another error because it tries to parse the "a" as an integer.

Reply:
Yes format error is one of the things and yesterday Tristan said this function will always get version in X.Y.z format so now 4 splits thing isn't required

Reply:
I checked this test for format error

Reply:
You'll have to check for that in the actual function then because this won't pass the test right now; it just throws an error:

```
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 11.27 s <<< FAILURE! - in UtilSpec
[ERROR] [util.groovy] getNextSnapshot error(UtilSpec)  Time elapsed: 0.264 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "a"
	at UtilSpec.[util.groovy] getNextSnapshot error(UtilSpec.groovy:437)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   UtilSpec.[util.groovy] getNextSnapshot error:437 » NumberFormat For input stri...
[INFO] 
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.758 s
[INFO] Finished at: 2020-07-16T02:43:27-04:00
[INFO] ------------------------------------------------------------------------
```

Use this command to run your tests: `mvn "-Dtest=UtilSpec#*getNextSnapshot*" test`.

Reply:
Yeah right it will throw the error in the function itself. Let me handle the case there itself, that if it's integer then only it should parse and change the 2nd test accordingly.